### PR TITLE
LPS-147072 Creating a kaleo process with ddm-image field causes NPE when viewing entries

### PR DESCRIPTION
**Pull request by jgarcialr**
Related Issue: https://issues.liferay.com/browse/LPS-147072

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/ImageDDMFormFieldValueRenderer.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.Locale;
+
+/**
+ * @author Bruno Basto
+ * @author Marcellus Tavares
+ */
+public class ImageDDMFormFieldValueRenderer
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/ImageDDMFormFieldValueRenderer.java:40]: Hi @jgarcialr, thanks for working on this!

I noticed that this code is the same from `DocumentLibraryDDMFormFieldValueRenderer`.
That said, do you mind creating a base class, like `BaseNumberDDMFormFieldValueRenderer` and `BaseTextDDMFormFieldValueRenderer`, so we can reuse the code in both fields (`DocumentLibrary` and `Image`)? We can call it `BaseFileDDMFormFieldValueRenderer`.

Thanks!

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/ImageDDMFormFieldValueRenderer.java (hunk 1 above)
Comment (jgarcialr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/ImageDDMFormFieldValueRenderer.java:40]: Hi @carolmariaabb , 
Sure, I'll change it now

Comment (jgarcialr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/render/ImageDDMFormFieldValueRenderer.java:22]: see 79677ad

**Comment by jgarcialr** [comment]
@carolmariaabb I've send the changes, hope it's ok

**Review by carolmariaabb — APPROVED** [review]

**Comment by jgarcialr** [comment]
Hi @carolmariaabb 

I was testing and see that ddm-image.ftl doesn't behave well with this renderer.
I think I'll have to change it, I did a supposition that ddm-image has same fields as ddm-document-library and was a wrong assumption.

**Comment by jgarcialr** [comment]
Please reject the pr and i'll perform changes and resend.

Sorry for the extra work :sweat_smile:

**Comment by carolmariaabb** [comment]
No problem, @jgarcialr .

Thank you for letting me know 😄.

**Pull request closed without merging**